pages/create-issue.tsx

```diff
@@ -1,48 +1,122 @@
 import { GetStaticProps } from 'next'
-import React, { useState } from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import BeproService from '../services/bepro';
 import GithubMicroService from '../services/github-microservice';
+import { setLoadingAttributes } from '../providers/loading-provider';
+import InputNumber from '../components/inputNumber';
+import { useRouter } from 'next/router';
 
-export default function PageCreateIssue() {
-
+interface Amount {
+  value?: string,
+  formattedValue: string,
+  floatValue?: number
+}
 
+export default function PageCreateIssue() {
   const [issueTitle, setIssueTitle] = useState("");
   const [issueDescription, setIssueDescription] = useState("");
-  const [issueAmount, setIssueAmount] = useState<number>(0);
+  const [issueAmount, setIssueAmount] = useState<Amount>({ value: '0', formattedValue: '0', floatValue: 0});
+  const [balance, setBalance] = useState<string>('0');
   const [allowedTransaction, setAllowedTransaction] = useState<boolean>(false);
+  const router = useRouter()
+
+  useEffect(() => {
+    let isMounted = true;
+    void async function getBeproBalance() {
+      await BeproService.login(); 
+      if (isMounted) {
```

Comment:
wouldn't this _always be true_ ? and wouldn't that make it so we're calling it on every update?

Reply:
This to prevent calling the state updater when the components unmount!

Reply:
As we are using an async promise call, we must use a reference variable to check already unmounted component for the next treatment of async response (avoiding memory leaks). When settled on an async function as a side-effect, this will happen on our [console debug](https://reactjs.org/docs/hooks-rules.html#eslint-plugin) when the component umount:
```console
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
```
Why? The memory leak will happen if the microservice took time to respond and the component was unmounted before the response was received. Though the component was unmounted, the response to the request will still be received on completion. And guess what will be triggered even when the component has been unmounted?
```jsx
// based on the suggestion you gave
useEffect(() => {
  async function getBalance() {
    await BeproService.login();
    return BeproService.network.getBEPROStaked();
  }
  
  // 👇 will be triggered even when the component has been unmounted.
  // Besides, updater state it's naturally async. I don't think use await here is necessary
  setBalance(await getBalance())
  
  return () => null;
}, [])
```
That's where I accomplished on the early suggestion. Let's wrapped it up:
```jsx
useEffect(() => {
  /* As we're calling an async function, using this to conditionally update
  *  component state prevents memory leak.
  */

  /* Everyhing here will be triggered only
  * on component mount. In this case,
  * we use a variable to reference when
  * the component mounts/umounts.
  * */
  let isMounted = true;

  void async function getBeproBalance() {
    await BeproService.login(); 

    // this will be only triggered if the component is mounted, free of wait for responses.
    if (isMounted) {
      setBalance(BeproService.network.getBEPROStaked());
    }
  }();

  /* Everyhing here will be triggered only
  * on component unmount. Set null here,
  * will just trigger nothing. In this case, we
  * set the reference as 'umount' (isMounted = false)
  * */
  return () => {
    isMounted = false;
  }
}, [])
```

Related: https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

Reply:
![image](https://user-images.githubusercontent.com/1077079/125058615-5af10200-e0a2-11eb-99f8-69998229d606.png)

https://user-images.githubusercontent.com/1077079/125058621-5d535c00-e0a2-11eb-8bfd-05202b8f7fe6.mp4